### PR TITLE
Initial commit on compatible API

### PR DESCRIPTION
**Pull request by jwhonce**
Signed-off-by: Jhon Honce <jhonce@redhat.com>

### CODE REVIEW

File: pkg/serviceapi/server.go [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+	"net"
+	"net/http"
+	"os"
+	"os/signal"
+	"syscall"
+	"time"
+
+	"github.com/containers/libpod/libpod"
+	log "github.com/sirupsen/logrus"
+
```
Comment (rhatdan) [inline pkg/serviceapi/server.go]: Remove line

File: pkg/serviceapi/server.go [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+	server := &http.Server{}
+	go func() {
+		<-quit
+		log.Debugf("HttpServer is shutting down")
+		ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+		defer cancel()
+		server.SetKeepAlivesEnabled(false)
+		if err := server.Shutdown(ctx); err != nil {
+			log.Panicf("cannot gracefully shut down the http server: %s", err)
```
Comment (rhatdan) [inline pkg/serviceapi/server.go]: Shouldn't we just log.Errorf?

File: pkg/serviceapi/server.go [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+		}
+		io.WriteString(w, string(buffer))
+	})
+	return &HttpServer{server, done, listeners[0]}, nil
+}
+
+func (s *HttpServer) Serve() error {
+	err := http.Serve(s.listener, nil)
+	if err != nil {
+		log.Panicf("Cannot start server: %s", err)
```
Comment (rhatdan) [inline pkg/serviceapi/server.go]: Return errors.Wrapf

File: pkg/serviceapi/server.go [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+type HttpServer struct {
+	*http.Server
+	done     chan struct{}
+	listener net.Listener
+}
+
+func NewServer(runtime *libpod.Runtime) (*HttpServer, error) {
+	listeners, err := activation.Listeners()
+	if err != nil {
+		log.Panicf("Cannot retrieve listeners: %s", err)
```
[the remark this answers is not in the document]
Comment (jwhonce) [reply pkg/serviceapi/server.go]: Panic is easier for now.  Will move to errors once we wire everything in.

**Pull request merged**